src/Centeva.ObjectStorage/Extensions/ObjectStorageExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace Centeva.ObjectStorage;
+
+public static class ObjectStorageExtensions
+{
+    public static async Task CopyAsync(this IObjectStorage sourceStorage, StoragePath sourcePath, IObjectStorage targetStorage, StoragePath? targetPath, CancellationToken cancellationToken = default)
+    {
+        var destPath = targetPath?.IsFolder == true
+            ? new StoragePath(StoragePath.Combine(targetPath.Full, sourcePath.Name))
+            : targetPath ?? sourcePath;
+
+        using var sourceStream = await sourceStorage.OpenReadAsync(sourcePath, cancellationToken);
```

Comment:
CopyAllAsync is checking for null values here, but this one assumes you get a Stream back